wagtailio/packages/views.py

```diff
@@ -0,0 +1,61 @@
+import requests
+from django.shortcuts import render
+from django.views import View
+from wagtail.admin import messages
+
+from .models import Grid, Package
+
+
+def process(url="https://djangopackages.org/api/v4/grids/?q=wagtail"):
+    gird_data = requests.get(url).json()
+    for item in gird_data.get("results", []):
+        title = item.get("title", "")
+        if "wagtail" in title.lower():
+            defaults = {key: item[key] for key in ["title", "slug", "description"]}
+            grid, _ = Grid.objects.update_or_create(
+                uid=item.get("id"), defaults=defaults
+            )
+            for idx, url in enumerate(item.get("packages", [])):
+                package_data = requests.get(url).json()
+                defaults = {
+                    key: package_data[key]
+                    for key in [
+                        "title",
+                        "slug",
+                        "repo_url",
+                        "pypi_version",
+                        "repo_forks",
+                        "repo_description",
+                        "pypi_url",
+                        "documentation_url",
+                        "repo_watchers",
+                        "participants",
+                    ] if not package_data[key] is None
+                }
+                defaults.update({"order": idx})
+                package, _ = Package.objects.update_or_create(
+                    uid=package_data.get("id"), defaults=defaults
+                )
+                grid.packages.add(package)
+
+    url = gird_data.get("next")
+    if url:
+        process(url)
```

Comment:
wagtail.io runs on Heroku which has a hard requestion timeout limit of 30s. How fast does this run?

suggestion: create a management command that can run on cron (say weekly)? Management commands do not have the limit

Reply:
Not within 30 sec. ;)

Did you know that Heroku returns a time out, but finishes the process? Not nice UX, getting a timeout, but workable if you know what happens.

I'm not sure if I prefer a job or manual process. I think we always want to check the result before we publish it.

Reply:
> Did you know that Heroku returns a time out, but finishes the process? Not nice UX, getting a timeout, but workable if you know what happens.

I did, but not many do. Probably not an issue wagtail.io specifically

Reply:
Fixed this with a help text. Also added a management command. So we can choose what to use.